src/Scaffolding/VS.Web.CG.Mvc/BlazorIdentity/BlazorIdentityGenerator.cs

```diff
@@ -0,0 +1,544 @@
+            }
+
+            var applicationUserModel = new IdentityApplicationUserModel
+            {
+                UserClassName = blazorIdentityModel.UserClassName,
+                UserClassNamespace = blazorIdentityModel.UserClassNamespace
+            };
+
+            if (createUserClass)
+            {
+                ExecuteApplicationUserTemplate(applicationUserModel);
+            }
+
+            if (createDbContext)
+            {
+                var dbContextModel = new IdentityDbContextModel
+                {
+                    DbContextName = blazorIdentityModel.DbContextName,
+                    DbContextNamespace = blazorIdentityModel.DbContextNamespace,
+                    UserClassModel = applicationUserModel
+                };
+
+                ExecuteDbContextTemplate(dbContextModel);
+                var connectionStringsWriter = new ConnectionStringsWriter(AppInfo, FileSystem);
+                connectionStringsWriter.AddConnectionString("DefaultConnection", dbContextModel.DbContextName, commandlineModel.DatabaseProvider);
+            }
+
+            //gather blazor identity files to generate
+            blazorIdentityModel.FilesToGenerate = new List<string>();
+            if (!string.IsNullOrEmpty(commandlineModel.Files))
+            {
+                blazorIdentityModel.FilesToGenerate = commandlineModel.Files.Split(IdentityHelper.SemicolonSeparator, StringSplitOptions.RemoveEmptyEntries).ToList();
+            }
+            else if (!string.IsNullOrEmpty(commandlineModel.ExcludeFiles))
+            {
+                IEnumerable<string> excludedFiles = commandlineModel.ExcludeFiles.Split(IdentityHelper.SemicolonSeparator, StringSplitOptions.RemoveEmptyEntries).Select(p => p.Trim()).ToList();
+                //validate excluded files
+                var errors = new List<string>();
+                var invalidFiles = excludedFiles.Where(f => !AllBlazorIdentityFiles.ContainsKey(f));
+                if (invalidFiles.Any())
+                {
+                    errors.Add(MessageStrings.InvalidFilesListMessage);
+                    errors.AddRange(invalidFiles);
+                }
+
+                if (errors.Count != 0)
+                {
+                    throw new InvalidOperationException(string.Join(Environment.NewLine, errors));
+                }
+
+                //get files to overwrite
+                blazorIdentityModel.FilesToGenerate = AllBlazorIdentityFiles.Keys.Except(excludedFiles).ToList();
+            }
+
+            if (blazorIdentityModel.FilesToGenerate.Count == 0)
+            {
+                blazorIdentityModel.FilesToGenerate = AllBlazorIdentityFiles.Keys.ToList();
+            }
+
+            blazorIdentityModel.BaseOutputPath = Path.Combine(AppInfo.ApplicationBasePath, commandlineModel.RelativeFolderPath);
+            return blazorIdentityModel;
+        }
+
+        private void ShowFileList()
+        {
+            Logger.LogMessage("File List:\n");
```

Comment:
This might be just my lack of knowledge on this codebase, but why are some things written to Logger and others (like down in ModifyFilesAsync) written to ConsoleLogger?

Reply:
It's needed for the DocumentBuilder class. Ideally there should only be one logger but this created earlier for dotnet-msidentity (DocumentBuilder was written alongside). Changed all references of ConsoleLogger to Logger other than the one needed for DocumentBuilder. These should be combined in our rework soon.